### ISSUE TITLE
Invalid credentials error on machineDeployment updates

### ISSUE THREAD

**Issue by prashanth26** [gardener/machine-controller-manager#542]
**What happened**:
Invalid Credentials error on rolled out machines

I am seeing this occurring when a machineDeployment has undergone a rolling update (with new machineset/machines) and is pointing to a new set of machineClass (and credentials) and one of the old machine pointing to the old machineClass (and old credentials) is stuck in the deletion and the credentials are outdated. Now with updating the credentials to a new one, only the new machineClass receives the update but the old machineClass doesn't. This seems to be a corner case that we need to improve handling here. 

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

- Create a machineDeployment with valid credentials.
- Now invalidate the credentials by removing access to it.
- Now trigger a rolling update in parallel by modifying the machineClass. 
- The update fails.
- Now update credentials to the new one on machineDeployment by pointing to new credentials. 
- Now the old machine is still stuck in deletion with old credentials. 

**Anything else we need to know**:
Probably occurs more with clusters in hibernation deletion. 

**Environment**:
Openstack

**Comment by rfranzke** [comment]
I think it makes more sense to move the issue to g/g as the generic worker actuator seems to be the culprit, i.e., it does not update old machine class secrets.
Do you agree?

**Comment by hardikdr** [comment]
Yes, I'd agree, if all the related secrets underneath are updated with correct creds, this issue is better resolved, we should move it to the g/g.
